daemon/k8s_watcher.go

```diff
@@ -67,51 +71,75 @@ func init() {
 	}
 }
 
-// k8sErrorHandler handles the error messages on a non verbose way by omitting
-// same error messages for a timeout defined with k8sErrLogTimeout.
+// k8sErrorUpdateGetUnmuteTime ensures an entry exists for this error string in
+// k8sErrMsg and returns the absolute time after which it can be printed again.
+// It updates the deadline on each call and assumes we only call it when we see
+// a particular error message.
+// On creating a new entry in the map, the returned deadline is the Epoch
+// start. This will always be before now.
+func k8sErrorUpdateGetUnmuteTime(errstr string, now time.Time) (unmuteDeadline time.Time) {
+	k8sErrMsgMU.Lock()
+	defer k8sErrMsgMU.Unlock()
+
+	unmuteDeadline, ok := k8sErrMsg[errstr]
+	if !ok {
+		k8sErrMsg[errstr] = now.Add(k8sErrLogTimeout)
+		return time.Unix(0, 0)
+	}
+
+	k8sErrMsg[errstr] = unmuteDeadline.Add(k8sErrLogTimeout)
+	return unmuteDeadline
```

Comment:
I don't think you wanted to do this because `Add` doesn't add the time to the receiver, it returns a new value. For this reason you should return the assigned value.
```
	k8sErrMsg[errstr] = unmuteDeadline.Add(k8sErrLogTimeout)
	// k8sErrMsg[errstr] == unmuteDeadline // since it's false we need to return the assigned value
        return k8sErrMsg[errstr]
```

Reply:
I had meant to return the original, looked up value but my logic was wrong as you mentioned above. I wasn't doing the rate-limiting correctly. This is fixed up.